src/usage/thunks/index.ts

```diff
@@ -0,0 +1,71 @@
+// Libraries
+import {Dispatch} from 'react'
+
+// API
+import {
+  getBillingAccount,
+  getBillingDate as apiGetBillingDate,
+  getHistory as apiGetHistory,
+} from 'src/usage/api'
+
+// Actions
+import {
+  setAccount,
+  setAccountStatus,
+  setBillingDate,
+  setBillingDateStatus,
+  setHistory,
+  setHistoryStatus,
+  Action,
+} from 'src/usage/reducers'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+export const getBillingDate = async (dispatch: Dispatch<Action>) => {
```

Comment:
Was wondering about your thoughts on extracting the repitition out?
```
const process = async (cbSetStatus, cbGet, cbSet) => {
    try {
        dispatch(cbSetStatus(RemoteDataState.Loading))
        const resp = await cbGet()

        if (resp.status !== 200) {
          throw new Error(resp.data.message)
        }
        dispatch(cbSet({...resp.data, status: RemoteDataState.Done}))
      } catch (error) {
        dispatch(cbSetStatus(RemoteDataState.Error))
        console.error(error)
      }
}
```

Reply:
I hear what you're getting at. I think that we might end up constraining ourselves later on with an API that's increasingly confined towards a specific pattern of expectations.